robot-name/robot-name_test.spec.js

```diff
@@ -12,9 +12,17 @@ describe("Robot", function() {
   });
 
   it("different robots have different names", function() {
-    var robotOne = new Robot();
-    var robotTwo = new Robot();
-    expect(robotOne.name).not.toEqual(robotTwo.name);
+    var usedNames= {};
+
+    var i = 0,
+        max = 10000;
+
+    for (i; i < max; i++) {
```

Comment:
Is it common to do the variable declaration outside of the loop? In the languages I work in, the preference is usually to declare the variable as close to where it is used as possible. I would also hard-code the max instead of assigning it to a variable here.